src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,109 @@ function updateStatsCheckBox() {
     $('input[id=statsShownCheckBox]').val($('#show-stats-checkbox').is(':checked'));
 }
 
-function toggleCollapse(e, pans) {
-    var expand = 'Expand';
-    var collapse = 'Collapse';
-    var panels = pans || $('div.panel-collapse');
-
-    if ($(e).html().trim().startsWith(expand)) {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
-            }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+/**
+ * Expands or collapses all panels when collapse/expand panels button is clicked.
+ * @param {DOM} element - The element that was clicked to invoke {@code #expandOrCollapsePanels}
+ * @param {DOM} panels - The panels to be expanded/collapsed. not defined if {@param element} is
+ *         collapse panels button.
+ */
+function expandOrCollapsePanels(element, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    var targetPanels = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(STRING_EXPAND);
+
+    if (isElementAnExpandButton) {
+        expandPanels(element, targetPanels);
+        replaceButtonHtmlAndTooltipText(element, STRING_EXPAND, STRING_COLLAPSE);
```

Comment:
Erm they are different things in different div. Button is more like a remote to all the panels and they are not related at all. (If you open feedback result page,  you would see Expand Questions button and each question panel as not really related
![screen shot 2017-03-22 at 11 59 25 pm](https://cloud.githubusercontent.com/assets/16921967/24207366/9e1985d8-0f5b-11e7-9b0b-6923ffc455d1.png)
)